lib/src/dart3_suggestors/null_safety_prep/fn_prop_null_aware_call_suggestor.dart

```diff
@@ -0,0 +1,178 @@
+///
+/// **After:**
+///
+/// ```dart
+/// // This will require no changes during a null-safety migration.
+/// props.someCallback?.call(someValue);
+/// ```
+class FnPropNullAwareCallSuggestor extends RecursiveAstVisitor
+    with ClassSuggestor {
+  ResolvedUnitResult? _result;
+
+  @override
+  visitExpressionStatement(ExpressionStatement node) {
+    super.visitExpressionStatement(node);
+
+    if (node.expression is! BinaryExpression) return;
+
+    final relevantExprStatement =
+        _getPropFunctionExpressionBeingCalledConditionally(
+            node.expression as BinaryExpression);
+    final inlineBinaryExpr =
+        // This cast is safe due to the type checks within `_getPropFunctionExpressionBeingCalledConditionally`.
+        relevantExprStatement?.expression as BinaryExpression?;
+    if (inlineBinaryExpr == null) return;
+    final relevantFnExpr =
+        // This cast is safe due to the type checks within `_getPropFunctionExpressionBeingCalledConditionally`.
+        inlineBinaryExpr.rightOperand as FunctionExpressionInvocation;
+    // This cast is safe due to the type checks within `_getPropFunctionExpressionBeingCalledConditionally`.
+    final fn = relevantFnExpr.function as PropertyAccess;
+
+    yieldPatch(
+        '${fn.target}.${fn.propertyName}?.call${relevantFnExpr.argumentList};',
+        node.offset,
+        node.end);
+  }
+
+  @override
+  visitIfStatement(IfStatement node) {
+    super.visitIfStatement(node);
+
+    if (node.condition is! BinaryExpression) return;
+
+    final relevantFnExprStatement =
+        _getPropFunctionExpressionBeingCalledConditionally(
+            node.condition as BinaryExpression);
+    final relevantFnExpr =
+        // This cast is safe due to the type checks within `_getPropFunctionExpressionBeingCalledConditionally`.
+        relevantFnExprStatement?.expression as FunctionExpressionInvocation?;
+    if (relevantFnExpr == null) return;
+    // This cast is safe due to the type checks within `_getPropFunctionExpressionBeingCalledConditionally`.
+    final fn = relevantFnExpr.function as PropertyAccess?;
+    if (fn == null) return;
+
+    yieldPatch(
+        '${fn.target}.${fn.propertyName}?.call${relevantFnExpr.argumentList};',
+        node.offset,
+        node.end);
+  }
+
+  /// Returns the function expression (e.g. `props.onClick(event)`) being called
+  /// after the null condition is checked.
+  ExpressionStatement? _getPropFunctionExpressionBeingCalledConditionally(
+      BinaryExpression condition) {
+    final parent = condition.parent;
+    if (parent is! IfStatement) return null;
+
+    final propFunctionBeingNullChecked =
+        _getPropFunctionBeingNullChecked(condition);
+    final ifStatement = parent;
+    if (ifStatement.elseStatement != null) return null;
+    if (ifStatement.parent
+            ?.thisOrAncestorOfType<IfStatement>()
+            ?.elseStatement !=
+        null) {
+      // There is an else-if statement present
+      return null;
```

Comment:
I think this check isn't as specific as it should be, and due to the `thisOrAncestorOfType`, might cause cases to get skipped if they're nested. For example, I'd expect this to get skipped with the current logic
```dart
if (condition) {
  if (props.onClick != null) props.onClick(e)
} else {}
```

Could we just check the parent directly instead?
```suggestion
    if (ifStatement.parent.tryCast<IfStatement>()?.elseStatement ==
        ifStatement) {
      // ifStatement is an else-if
      return null;
    }
```